parquet-hadoop/src/main/java/parquet/hadoop/MemoryManager.java

```diff
@@ -47,6 +47,7 @@
   private final long minMemoryAllocation;
   private final Map<InternalParquetRecordWriter, Long> writerList = new
       HashMap<InternalParquetRecordWriter, Long>();
+  private final Map<String, Runnable> callBacks = new HashMap<String, Runnable>();
```

Comment:
Do you want to use a `Callable<MemorManager>` or `Callable<MemoryManagerInfo>` or something like that so the callable can be passed things like the scale value etc.?